workflows: `issue_comment` triggers refactoring

nbusseneau opened the pull request:
Please review per commit :)

nbusseneau committed:
workflows: reformat `issue_comment`-based checks

Ensure all checks use a consistent format.

Signed-off-by: Nicolas Busseneau <nicolas@isovalent.com>

nbusseneau committed:
workflows: add `/test` and `/`-prefixed triggers

For now keep the previous and new triggers as a transition phase.

Signed-off-by: Nicolas Busseneau <nicolas@isovalent.com>

twpayne commented on .github/workflows/conformance-aks-v1.10.yaml:
```diff
@@ -53,10 +53,10 @@ concurrency:
-        startsWith(github.event.comment.body, '/ci-aks-1.10') ||
-        startsWith(github.event.comment.body, 'test-backport-1.10') ||
-        startsWith(github.event.comment.body, '/test-backport-1.10')
+        github.event.comment.body == 'ci-aks-1.10' ||
```
Doing an exact match here will break if there's a newline after `/ci-aks-1.0`. Can we use a regexp match instead?

twpayne replied:
It seems that [there's no regexp matching available](https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#functions) 😢

twpayne replied:
As `/test` is the problem, how about for `/test` only:

```
(startsWith(github.event.comment.body, '/test') && !startsWith(github.event.comment.body, '/test-'))
```

it's a hack, but is resistant to new lines and comments after the `/command`.

nbusseneau replied:
Yea, as said in the commit message my main goal was getting rid of the "double testing" issue, with this limitation as side-effect. Unfortunately GitHub Actions does not support filtering strings using regex (see [here](https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions) for what's possible in native syntax), so even if we wanted to we couldn't do it from within the `if` context. Would have to kludge around with a shell step actively failing the job if not matching...

nbusseneau replied:
You posted so fast GH API did not have time to show me the messages, wow! :D
OK let's ship it.

nbusseneau replied:
PTAL.

nbusseneau committed:
workflows: harden `startsWith` checks

This will avoid triggering both regular and 1.10 CI when commenting
`/test-backport-1.10` on a backport PR, while still allowing additional
comments or newlines after the initial test trigger phrase.

Signed-off-by: Nicolas Busseneau <nicolas@isovalent.com>

twpayne reviewed (approved):
LGTM

We also need to add docs, but that can be in a follow-up PR.

nbusseneau committed:
update MLH config with new `/test` trigger

Signed-off-by: Nicolas Busseneau <nicolas@isovalent.com>

nbusseneau committed:
documentation: update trigger phrases

- `test-me-please` replaced with `/test`.
- `build-me-please` replace with `/build`.
- Other triggers as-is but prefixed with `/`.
- Unified race detection trigger phrases with the rest of the document:
  singular race detection jobs will expose their unique trigger phrases
  directly from the PR checks.

Signed-off-by: Nicolas Busseneau <nicolas@isovalent.com>

nbusseneau commented on .github/maintainers-little-helper.yaml:
```diff
@@ -50,7 +50,7 @@ flake-tracker:
   jenkins-config:
     jenkins-url: https://jenkins.cilium.io
-    regex-trigger: (^test-me-please)
+    regex-trigger: (^/?test(-me-please)?)
```
@aanm Will that work as intended?

aanm replied:
@nbusseneau yes but the right regex would be `(^(/test|test-me-please))` otherwise `/test-me-please` will also be considered.

nbusseneau replied:
Yes, I thought it was nice having it in during the transition phase. I also used that same regex allowing both on Jenkins.

pchaigno commented on .github/workflows/conformance-aws-cni.yaml:
```diff
@@ -108,15 +108,15 @@ jobs:
-        (startsWith(github.event.comment.body, 'test-me-please') && (needs.check_changes.outputs.tested == 'true'))
-      )) ||
+        startsWith(github.event.comment.body, 'test-me-please')
+      ) && needs.check_changes.outputs.tested == 'true') ||
```
If I'm reading this correctly, it's a change in behavior. `ci-awscni` now won't trigger the test if only `test/` or `Documentation/` were modified. That was done on purpose, as I'm assuming someone writing `ci-awscni` explicitly know what they are doing.

That's useful for quick tests where you only want to send a nop change and trigger the tests.

nbusseneau replied:
Indeed, I didn't think this behaviour was useful so I erred on the side of simplifying the conditions for consistency. If we think it is, I can add it back in :)

pchaigno replied:
Unless it's adding too much complexity, yes please :pray:

nbusseneau replied:
PTAL.

pchaigno replied:
I'm not seeing any change :detective: Was it adding too much complexity?

nbusseneau replied:
Uuuh, is the GitHub cache playing tricks on you? 👀 Anyway, the changes are available here, from the force push link below: https://github.com/cilium/cilium/compare/f8baa22cbe9fb6a943f7bdfbf04dc31ae0ca349c..e44285b0bfc7c423921b07460e0a58461fecd7ec

pchaigno replied:
Uh, why are they in the documentation commit? Might be worth updating the commit title at least.

nbusseneau replied:
Oh are they? I mixed up the fixup then.

nbusseneau replied:
I have reworked the commits so that the changes are in the proper place.

pchaigno commented on .github/workflows/conformance-aks.yaml:
```diff
@@ -56,10 +56,10 @@ concurrency:
-        startsWith(github.event.comment.body, '/test')
+        (startsWith(github.event.comment.body, 'ci-aks') && !startsWith(github.event.comment.body, 'ci-aks-')) ||
+        (startsWith(github.event.comment.body, '/ci-aks') && !startsWith(github.event.comment.body, '/ci-aks-')) ||
+        (startsWith(github.event.comment.body, 'test-me-please') && !startsWith(github.event.comment.body, 'test-me-please-')) ||
```
Is this required for `test-me-please`? We don't have `test-me-please-1.10`, do we?

nbusseneau replied:
Not required, just added via grep replace. Decided to leave it because it has no side-effects and will be removed anyway as this is only for the transition period.

pchaigno reviewed (changes requested):
One concern below, the other is a nit. Also, where was this tested?

pchaigno reviewed (approved):
@nbusseneau Did you validate this with a test commit already?
